packages/sit-onyx/src/composables/useCustomValidity.ts

```diff
@@ -53,6 +55,29 @@ export type FormErrorMessages = {
   longMessage?: string;
 };
 
+/**
+ * Transforms a customError into the format needed to display an error preview and extended message
+ */
+export const getCustomErrors = (customError?: CustomErrorType): FormErrorMessages | undefined => {
+  if (!customError) return;
+  if (typeof customError === "string") {
+    // we can't guarantee a custom error message will be short,
+    // so in case it overflows, by adding it to "longMessage",
+    // it will still be visible in a tooltip
+    return { shortMessage: customError, longMessage: customError };
+  }
+  return customError;
+};
+
+export const getCustomErrorText = (customError?: CustomErrorType): string | undefined => {
+  if (!customError) return;
+  if (typeof customError === "string") {
+    return customError;
+  }
```

Comment:
Can't we also use `getCustomErrors()` here?

Reply:
then we'll have to check if shortMessage + longMessage are the same else you would have

`getCustomErrorText('foo')` return `"foo: foo"`